src/app/course/course-challenge-snippet/course-challenge-snippet.component.html

```diff
@@ -1,104 +1,243 @@
-                            Grade: {{ challenge.total_event_grade | number: '1.0-1' || 'No Grade'}}%
-                        </ng-template>
-                    </div>
-                    <div class="linkBtn">
-                        <ng-container *ngIf="challenge.is_allowed_to_open || user?.is_teacher; else notAllowedOpen">
-                            <a [routerLink]="['/course', course.id, 'event', challenge.id]" appearance="primary" class="tui-form__button"
+                        <div class="dates">
+                            <div>From: {{ challenge.start_date | date: 'EEEE, MMM d, y, h:mm a' }}</div>
+                            <div>Until: {{ challenge.end_date | date: 'EEEE, MMM d, y, h:mm a' }}</div>
+                        </div>
+                        <div class="availability">
+                            <tui-tag *ngIf="challenge.is_open; else challengeComingSoon" size="m" status="success"
+                                     value="Available"></tui-tag>
+                            <ng-template #challengeComingSoon>
+                                <tui-tag size="m" status="error" value="Coming Soon"></tui-tag>
+                            </ng-template>
+                        </div>
+                        <div class="grade">
+                            <ng-container *ngIf="challenge.is_not_available_yet ; else totalEventGrade">
+                                Grade: TBA
+                            </ng-container>
+                            <ng-template #totalEventGrade>
+                                Grade: {{ challenge.total_event_grade | number: '1.0-1' || 'No Grade'}}%
+                            </ng-template>
+                        </div>
+                        <div class="linkBtn">
+                            <ng-container *ngIf="challenge.is_allowed_to_open || user?.is_teacher; else notAllowedOpen">
+                                <a [routerLink]="['/course', course.id, 'event', challenge.id]" appearance="primary"
+                                   class="tui-form__button"
+                                   size="s"
+                                   tuiButton
+                                >
+                                    {{ user.is_teacher ? 'View' : 'Complete' }}
+                                </a>
+                            </ng-container>
+                            <ng-template #notAllowedOpen>
+                                <a [disabled]="true" appearance="primary" aria-label="Complete" class="tui-form__button"
+                                   href=""
+                                   size="s"
+                                   tuiButton>
+                                    Complete
+                                </a>
+                            </ng-template>
+                            <a *ngIf="challenge.has_edit_permission"
+                               [routerLink]="['/course', course.id, 'new-event', challenge.id]"
```

Comment:
Why is the url `new-event` is that's edit?

Reply:
I am pretty sure that is how Lydia coded it

Reply:
Yes, it is. This is her code in master:
https://github.com/COSC-447-Directed-Stuides/canvas-gamification-ui-COSOC447-directed-studies/blob/master/src/app/course/course-challenge-snippet/course-challenge-snippet.component.html